### DISCUSSION TITLE
Muz çiftçisi sorusu

### DISCUSSION THREAD

**Discussion by sonsuzus** [sonsuzus/sonsuzus#100] [Zeka Sorusu]
Çiftçinin biri muz yetiştiriyor. Mevsim sonunda 3000 kg muz elde etmiş. Bu muzları kasabada satacak. Yalnız en yakın kasaba 1000 km uzakta. Çiftçi kasabaya muzları hantal ve obur filine taşıttıracak. Ancak hantal ve obur fil kilometre başına 1 kg muz yiyor ve aynı anda 1000 kg’dan fazla muz taşıyamıyor. Çiftçi kasabaya en fazla kaç kilo muz götürebilir?
İstediği noktalara muzları bırakıp gidip gelerek kasabaya hiç muz götürebilir mi?
Not: Çiftçinin kasabadan köyüne filiyle dönmesine gerek yok, çiftçi, yaşlı filini de satıp, muzların satımından elde ettiği parayla daha genç ve daha az muz yiyen ekonomik bir fil alabilir.

**Comment by PurpleResul** [comment]
bunu python kodu şeklinde yazarsak;
muz = 3000       # Başlangıçtaki muz miktarı (kg)
filkapasite = 1000     # Filin taşıma kapasitesi (kg)
muztüketim= 1 # Filin kilometre başına muz tüketimi (kg/km)
mesafe = 1000    # Kasabaya olan mesafe (km)
mesafe = toplam mesafe

while mesafe > 0:
    # Her 1000 kg muz taşındığında, fil her kilometrede 1 kg muz yiyor.
    # Filin taşıma kapasitesi 1000 kg, dolayısıyla her 1000 kg'ı 1 km'de taşımak için harcanan muz miktarını hesapla.
    muz_tuketimi = filkapasite * muztüketim* (mesafe / toplam_mesafe)
    muz -= muz_tuketimi  # Taşınan muz miktarını güncelle
    
    # Kasabaya muz taşıyabilmek için bu taşıma stratejisinin uygun olup olmadığını kontrol et
    if muz <= 0:
        print("taşıma gerçekleşemedi")
        break  # Muz kalmazsa taşıma mümkün olmaz

    mesafe -= 1  # 1 km ilerleme

print(muz)

bu kodları çalıştırınca sonuç "taşıyamaz" çıkıyor yani yolda yaşlı fil bütün muzu yer.

**Reply by sonsuzus** [reply]
Ama taşıyabiliyor. Demek ki algoritma yanlış.

**Reply by PurpleResul** [reply]
hocam yanlış yazmışım da ama yani sonuçta fil yolda bütün muzları yemez mi?

**Reply by sonsuzus** [reply]
Hepsini yemez. eğer taşıma işlemini doğru yaparsan bir kısmını taşıyor.

**Comment by Sevvalgocgun1** [comment]
def max_muz():
    muz = 3000
    mesafe = 1000
    kapasite = 1000
    tuketim = 1
    
    # İlk olarak 3 tur yapmamız gerekiyor
    toplam = 0
    while muz >= kapasite:
        muz -= (mesafe * tuketim * 2)  # Gidip gelerek taşıma yapıyoruz
        toplam += kapasite
        
    return toplam

print(max_muz())
